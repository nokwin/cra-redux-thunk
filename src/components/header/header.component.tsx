import React from "react";
import { Link } from "react-router-dom";

import "./header.styles.scss";

export const Header = () => (
  <div className="header">
    <Link to="/" className="header--link-item">
      Homepage
    </Link>
    <Link to="/counter" className="header--link-item">
      Counter
    </Link>
  </div>
);
