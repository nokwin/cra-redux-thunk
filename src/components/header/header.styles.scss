.header {
  padding: 10px;

  &--link-item {
    margin-right: 20px;
  }
}
