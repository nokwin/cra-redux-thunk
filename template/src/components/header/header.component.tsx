import React from "react";
import { Link } from "react-router-dom";

import "./header.styles.scss";

export const Header = () => (
  <div className="header">
    <Link to="/" className="header__link-item">
      Homepage
    </Link>
    <Link to="/counter" className="header__link-item">
      Counter
    </Link>
    <Link to="/forms" className="header__link-item">
      Forms
    </Link>
    <Link to="/api" className="header__link-item">
      Api Example
    </Link>
  </div>
);
