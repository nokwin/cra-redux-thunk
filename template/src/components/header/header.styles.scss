.header {
  padding: 10px;

  &__link-item {
    margin-right: 20px;
  }
}
