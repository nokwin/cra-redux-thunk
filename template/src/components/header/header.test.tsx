import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import { Header } from "./header.component";

describe("Header component", () => {
  it("should render header component", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
