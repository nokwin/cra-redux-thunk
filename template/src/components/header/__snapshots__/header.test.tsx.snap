// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render header component 1`] = `
<div
  className="header"
>
  <a
    className="header__link-item"
    href="/"
    onClick={[Function]}
  >
    Homepage
  </a>
  <a
    className="header__link-item"
    href="/counter"
    onClick={[Function]}
  >
    Counter
  </a>
  <a
    className="header__link-item"
    href="/forms"
    onClick={[Function]}
  >
    Forms
  </a>
  <a
    className="header__link-item"
    href="/api"
    onClick={[Function]}
  >
    Api Example
  </a>
</div>
`;
